### PR TITLE
Feature/diagnostics level select

### PR DESCRIPTION
**Pull request by elynnwu**
## Purpose

This PR adds the capability of saving a vertical slice using `z_select` in the diagnostics config.

The expected syntax is as follows:
```
diagnostics_config:
  path: output
  output_format: netcdf
  z_select:
    - level: 1
       names:
         - ua
         - va
    - level: 10
       names:
         - pt  
```

## Code changes:

- diagnostics.py: added `ZSelect` class for storing selected vertical slice and the variables to be saved
   - fix column integral mistake, need to multiple by `1/gravity`

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://drive.google.com/file/d/1R0nqOxfYnzaSdoYdt8yjx5J482ETI2Ft/view?usp=sharing).
- [ ] Docstrings and type hints are added to new and updated routines, as appropriate
- [ ] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [ ] For each public change and fix in `pace-util`, HISTORY has been updated
- [ ] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: driver/pace/driver/diagnostics.py [hunk 1]
```diff
@@ -34,6 +36,35 @@ def cleanup(self):
         ...
 
 
+@dataclasses.dataclass
+class ZSelect:
+    level: int
+    names: List[str]
+
+    def slice_data(self, state: DycoreState):
```
Comment (twicki) [inline driver/pace/driver/diagnostics.py:44]: technically we select and don't slice here - not super relevant as selecting is a 1-sized slice, but if we actually wan to implement both we might want to rename it here already?

File: driver/pace/driver/diagnostics.py [hunk 2]
```diff
@@ -34,6 +36,35 @@ def cleanup(self):
         ...
 
 
+@dataclasses.dataclass
+class ZSelect:
+    level: int
```
Comment (twicki) [inline driver/pace/driver/diagnostics.py:41]: what is the main motivation behind 1 level for all names?

File: driver/pace/driver/diagnostics.py [hunk 3]
```diff
@@ -34,6 +36,35 @@ def cleanup(self):
...
+        output = {}
+        for name in self.names:
+            if name not in state.__dict__.keys():
+                raise ValueError(f"Invalid state variable {name} for level select")
+            assert len(getattr(state, name).dims) > 2
+            if getattr(state, name).dims[2] != (
+                pace.util.Z_DIM or pace.util.Z_INTERFACE_DIM
+            ):
+                raise ValueError(
+                    f"z_select only works for state variables with a z dimension. \
```
Comment (twicki) [inline driver/pace/driver/diagnostics.py:54]: we don't actually have a field that triggers this, right?

**Review by twicki — APPROVED** [review]
just a few questions

File: driver/pace/driver/diagnostics.py (hunk 1 above)
Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:44]: Changed it to select data.

File: driver/pace/driver/diagnostics.py (hunk 2 above)
Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:41]: That's the more intuitive way for me. Also, ZSelect implies that we are selecting by z level. So I think this makes sense.

File: driver/pace/driver/diagnostics.py (hunk 3 above)
Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:54]: Correct, while writing test I realized this is not possible based on the current DycoreState.

Comment (elynnwu) [reply driver/pace/driver/diagnostics.py:54]: Added test

**Pull request merged**